lms/djangoapps/learner_recommendations/views.py

```diff
@@ -99,20 +100,33 @@ def get(self, request, course_id):
             return Response(status=404)
 
         is_control = is_control if has_is_control else None
+        if is_control:
+            return Response(status=404)
```

Comment:
This response should not be a 404.
```suggestion
        if is_control or is_control is None:
            return Response(
                RecommendationsSerializer(
                    {
                        "program_upsell": None,
                        "courses": [],
                        "is_control": is_control,
                    }
                ).data,
                status=200,
            )
```

Reply:
Can you write a test for this case?